app/src/main/java/ch/epfl/sdp/drone3d/ui/map/offline/ManageOfflineMapActivity.kt

```diff
@@ -0,0 +1,147 @@
+package ch.epfl.sdp.drone3d.ui.map.offline
+
+import android.os.Bundle
+import android.widget.ProgressBar
+import android.widget.TextView
+import android.view.View
+import androidx.recyclerview.widget.RecyclerView
+import ch.epfl.sdp.drone3d.R
+import ch.epfl.sdp.drone3d.map.offline.OfflineMapSaver
+import ch.epfl.sdp.drone3d.map.offline.OfflineMapSaverImpl
+import ch.epfl.sdp.drone3d.ui.ToastHandler
+import ch.epfl.sdp.drone3d.ui.map.BaseMapActivity
+import com.google.android.material.floatingactionbutton.FloatingActionButton
+import com.mapbox.mapboxsdk.Mapbox
+import com.mapbox.mapboxsdk.maps.MapboxMap
+import com.mapbox.mapboxsdk.maps.OnMapReadyCallback
+import com.mapbox.mapboxsdk.maps.Style
+import com.mapbox.mapboxsdk.offline.OfflineRegion
+import com.mapbox.mapboxsdk.offline.OfflineRegionError
+import com.mapbox.mapboxsdk.offline.OfflineRegionStatus
+import timber.log.Timber
+import java.lang.StringBuilder
+import java.lang.System.currentTimeMillis
+import kotlin.math.min
+
+class ManageOfflineMapActivity : BaseMapActivity(), OnMapReadyCallback {
```

Comment:
A few words about the activity would be very much appreciated

Reply:
Yes you are right, it's done now